x-pack/plugins/security_solution/cypress/e2e/detection_rules/prebuilt_rules_install_update_workflows.cy.ts

```diff
@@ -45,35 +45,57 @@ describe('Detection rules, Prebuilt Rules Installation and Update workflow', ()
 
   describe('Installation of prebuilt rules package via Fleet', () => {
     beforeEach(() => {
-      cy.intercept('POST', '/api/fleet/epm/packages/_bulk*').as('installPackage');
+      cy.intercept('POST', '/api/fleet/epm/packages/_bulk*').as('installPackageBulk');
+      cy.intercept('POST', '/api/fleet/epm/packages/security_detection_engine/*').as(
+        'installPackage'
+      );
       waitForRulesTableToBeLoaded();
     });
 
     it('should install package from Fleet in the background', () => {
       /* Assert that the package in installed from Fleet by checking that
       /* the installSource is "registry", as opposed to "bundle" */
-      cy.wait('@installPackage', {
+      cy.wait('@installPackageBulk', {
         timeout: 60000,
-      }).then(({ response }) => {
-        cy.wrap(response?.statusCode).should('eql', 200);
-
-        const packages = response?.body.items.map(({ name, result }: BulkInstallPackageInfo) => ({
-          name,
-          installSource: result.installSource,
-        }));
-
-        expect(packages.length).to.have.greaterThan(0);
-        expect(packages).to.deep.include.members([
-          { name: 'security_detection_engine', installSource: 'registry' },
-        ]);
+      }).then(({ response: bulkResponse }) => {
+        cy.wrap(bulkResponse?.statusCode).should('eql', 200);
+
+        const packages = bulkResponse?.body.items.map(
+          ({ name, result }: BulkInstallPackageInfo) => ({
+            name,
+            installSource: result.installSource,
+          })
+        );
+
+        const packagesBulkInstalled = packages.map(({ name }: { name: string }) => name);
+
+        // Under normal flow the package is installed via the Fleet bulk install API.
+        // However, for testing purposes the package can be installed via the Fleet individual install API,
+        // so we need to intercept and wait for that request as well.
+        if (!packagesBulkInstalled.includes('security_detection_engine')) {
```

Comment:
It's okay to leave the logic here for now, but for a more robust solution, we should consider introducing an API endpoint on the detection engine side. This endpoint would handle all the complexities of package installation logic internally. We currently have duplication of this logic on both the frontend and backend. That could be part of Serverless: https://github.com/elastic/kibana/issues/150587.